File: ts/rot13.ts
// https://www.codewars.com/kata/52223df9e8f98c7aa7000062/train/typescript

export function rot13(str: string): string {
    return str.replace(/[A-Za-z]/g, (char) => {
      const base = char <= 'Z' ? 'A'.charCodeAt(0) : 'a'.charCodeAt(0);
      return String.fromCharCode(((char.charCodeAt(0) - base + 13) % 26) + base);
    });
  }

File: python/rot13.py
# https://www.codewars.com/kata/530e15517bc88ac656000716

def rot13(string):
    result = ''
    for char in string:
        if char.isalpha():
            if char.isupper():
                result += chr((ord(char) - 65 + 13) % 26 + 65)
            else:
                result += chr((ord(char) - 97 + 13) % 26 + 97)
        else:
            result += char
    return result
